mlir/lib/Dialect/MIOpen/utility/transformMapUtils.cpp

```diff
@@ -300,26 +308,40 @@ propagateUnmergeVectorization(T &&dimAndLength,
                               const VectorizationData &input) {
   Optional<VectorizationInfo> result;
   int64_t previousDimsStride = 1;
+  Optional<int64_t> previousAlign;
   for (auto pair : dimAndLength) {
     uint32_t upperDim = std::get<0>(pair);
     int64_t dimLength = std::get<1>(pair);
     if (input[upperDim].hasValue()) {
       VectorizationInfo upperInfo = *input[upperDim];
       if (!result.hasValue())
-        result = VectorizationInfo(1, upperInfo.needsCoefficient);
+        result = VectorizationInfo(1, upperInfo.needsCoefficient, 1);
       // Catch
       // 1. Dimensions merged out of order and then
       // 2. Previous dimensions having incomplete vector lengths
       if (upperInfo.needsCoefficient == previousDimsStride &&
           (result->maxLength * result->needsCoefficient) ==
               previousDimsStride) {
         result->maxLength *= upperInfo.maxLength;
+        if (!previousAlign.has_value())
+          previousAlign = upperInfo.alignment * previousDimsStride;
```

Comment:
Question: Why shall the alignment multiply the `previousDimsStride`? Could you give an example?

Reply:
Well, if the current dimension is aligned to 4 (that is, its initial value is divisible by 4) and the stride up to that point has been 3, we now have a value divisible by 12.

Now, since this value gets set all the time, it's fundamentally true that `previousDimsStride` is 1 here, but it makes sense to leave it in for symmetry's sake.